celltype_annotation/utils/singler-helper-functions.R

```diff
@@ -0,0 +1,88 @@
+# This function is used to create a stacked bar chart for all SingleR references of a given label type 
+# label type is either label.fine, label.main, or label.ont
+
+celltype_barchart <- function(celltype_assignments,
+                              label_type){
+  
+  top_celltypes <- celltype_assignments |>
+    dplyr::count(reference_factor, celltype_top) |>
+    tidyr::drop_na()
+  
+  ggplot(top_celltypes, aes(x = reference_factor, fill = celltype_top, y = n, label = n)) + 
+    geom_bar(position = "stack", stat = "identity") +
+    labs(
+      x = "Reference dataset",
+      y = "Number of cells", 
+      fill = "Cell type",
+      title = label_type
+    ) +
+    geom_text(size = 3, position = position_stack(vjust = 0.5)) +
+    guides(x = guide_axis(angle = 90)) 
+  
+}
+
+# create a table of the celltype assignments for each reference (e.g. HumanPrimaryCellAtlasData)
+celltype_table <- function(celltype_assignments){
+  
+  all_celltypes <- celltype_assignments |> 
+    dplyr::group_by(reference, celltype) |> 
+    dplyr::count() |> 
+    dplyr::ungroup() |> 
+    tidyr::spread(reference, n)
+
+  return(all_celltypes)
+
+}
+
+
+# creates a plot showing the distribution of the desired metric with reference on the x-axis
+# color by whether or not labels are pruned or not
+# facet by label type (main, fine, or ont)
+# we will use this function for looking at delta, median delta, and score distributions
+plot_metric_distribution <- function(all_results,
+                                     metric_type){
+  
+  metric_plot <- ggplot(all_results, aes_string(x = "reference", y = metric_type, group = "reference", color = "pruned")) +
```

Comment:
Looking at this again, I don't think `aes_string()` even needs to be tweaked..it doesn't seem to be needed at all? You literally are plotting a column called `reference`, right? Otherwise I'd expect a string to be passed into the function and used here. 

yet another untested suggestion but i think it will work fine (famous last words..)
```suggestion
  metric_plot <- ggplot(all_results, aes_string(x = reference, y = metric_type, group = reference, color = pruned)) +
```